lib/email_api/emailer.py

```diff
@@ -0,0 +1,147 @@
+import os
+from typing import Tuple, Dict, List, Optional
+from smtplib import SMTP_SSL
+from email.header import Header
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.utils import formatdate
+
+from structs.email_params import EmailParams
+from structs.smtp_account import SMTPAccount
+
+
+class Emailer:
+    """
+    Emailer class is responsible for building and sending emails.
+    """
+
+    # Holds absolute dirpath to directory where email attachements files are stored
+    EMAIL_ATTACHMENTS_ROOT_DIR = os.path.normpath(
+        os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), "../email_attachments"
+        )
+    )
+
+    def __init__(self, smtp_account: SMTPAccount):
+        self._smtp_account = smtp_account
+
+    def send_multiple_emails(
+        self, emails: Dict[Tuple[str], EmailParams], as_html: bool = True
+    ):
+        """
+        send multiple emails
+        :param emails: (Dict) keys are list of email addresses, values are configured EmailParams dataclass
+        :param as_html: whether to send emails using html message body (if True), or plaintext (if False) (Default True)
+        """
+        for email_to, email_params in emails.items():
+            self.send_email(email_to, email_params, as_html)
+
+    def send_email(
+        self, email_to: Tuple[str], email_params: EmailParams, as_html: bool = True
+    ):
+        """
+        send a single email
+        :param email_params: An EmailParams Dataclass which holds all config info needed to build and send an email
+        :param email_to: email addresses to send the email to
+        :param as_html: whether to send email using html message body (if True), or plaintext (if False) (Default True)
+
+        """
+        send_to = email_params.email_from
+        if email_params.email_cc:
+            send_to += ", "
+            send_to += ", ".join(email_params.email_cc)
+
+        msg = Emailer._build_message(email_params, email_to, as_html)
+        with SMTP_SSL(
```

Comment:
We should add logging here, to track what exact calls are being made in-case we get blamed for something over SMTP